lib/temporal/client.rb

```diff
@@ -398,13 +398,23 @@ def fail_activity(async_token, exception)
     #
     # @return [Temporal::Workflow::History] workflow's execution history
     def get_workflow_history(namespace:, workflow_id:, run_id:)
-      history_response = connection.get_workflow_execution_history(
-        namespace: namespace,
-        workflow_id: workflow_id,
-        run_id: run_id
-      )
+      next_page_token = nil
+      events = []
+      loop do
+        response =
+          connection.get_workflow_execution_history(
+            namespace: namespace,
+            workflow_id: workflow_id,
+            run_id: run_id,
+            next_page_token: next_page_token,
+          )
+        events.concat(response.history.events.to_a)
+        next_page_token = response.next_page_token
+
+        break if next_page_token.empty?
+      end
 
-      Workflow::History.new(history_response.history.events)
+      Workflow::History.new(response.history.events)
```

Comment:
Did you mean to create the workflow history object from the `events` property? Creating it from `response.history.events` is incorrect.